Detectors/MUON/MCH/Raw/Decoder/include/MCHRawDecoder/DataDecoder.h

```diff
@@ -47,10 +47,51 @@ struct OrbitInfoHash {
 class DataDecoder
 {
  public:
+  struct SampaInfo {
+    union {
+      uint32_t id = 0;
+      struct {
+        uint32_t chip : 1;
+        uint32_t ds : 6;
+        uint32_t solar : 16;
+        uint32_t unused : 9;
+      };
+    };
+
+    union {
+      // default value
+      uint64_t time = 0x0000000000000000;
+      struct {                       ///
+        uint32_t sampaTime : 10;     /// bit 0 to 9: sampa time
+        uint32_t bunchCrossing : 20; /// bit 10 to 29: bunch crossing counter
+        uint32_t reserved : 2;       /// bit 30 to 31: reserved
+        uint32_t orbit;              /// bit 32 to 63: orbit
+      };                             ///
+    };
+    uint32_t getBXTime()
+    {
+      return (bunchCrossing + (sampaTime * 4));
+    }
+  };
+
+  struct TimeFrameInfo {
```

Comment:
Is there a reason to have a custom struct here and not use directly e.g. IntereactionRecord from https://github.com/AliceO2Group/AliceO2/blob/dev/DataFormats/common/include/CommonDataFormat/InteractionRecord.h ?

Reply:
Good suggestion. I think there is no reason not to use an existing structure.

Reply:
well, actually, on closer look, the issue might be that the InteractionRecord's bunch crossing is 16 bits, while the DS bx is 20 bits, right ?

Reply:
Yes, indeed one needs 20 bits for storing the SAMPA bunch crossing counter.